### ISSUE TITLE
[FEATURE] Wiktionary Database

### ISSUE THREAD

**Issue by nimeska** [StrangeGirlMurph/obsidian-wikipedia-search#33]
Hello! Is there any chance that you would add some kind of Wiktionary extension? To be able to search and link, like with Wikipedia. 
I don't have a lot of knowledge about coding, so if that would be hard to do, no problem and sorry to bother:)

**Comment by StrangeGirlMurph** [comment]
Hey! Thank you so much for reaching out. It's great that you had the courage to ask. After a quick Google search it seems like this is technically possible. I will have to look further into it though. Before implementing anything I need to know how exactly you plan on using it. I never use Wiktionary and don't really know how single pages are generally build up. What use case do you have in mind? Do want whole pages? Just simple definitions? All languages?

**Comment by StrangeGirlMurph** [comment]
Just simple hyperlinking of words to their Wiktionary pages won't be a problem I think and I am happy to implement that. The question is if you need more. Do you also want the data of a Wiktionary page itself like with the {intro} for Wikipedia articles and so on? If yes the question is what exactly is worth retrieving for a Wiktionary entry what if that is technically possible to get.

**Comment by codey-badger** [comment]
Hi everyone!

I agree with nimeska; it would be super useful even with just a few template tags like {url}. Thinking about it, what else could be useful to get? Each section can be quite extensive, and there are many languages for some words.

Let me address some of your previous questions from my point of view (English is not my native language, so sorry if I'm not completely clear):

**"What use case do you have in mind?"**

- I want to leverage linking features from Obsidian to see specific words in different contexts and have the link in the front matter accessible to the full meaning.

**"Do you want whole pages?"**

- I think whole pages wouldn't be viable because they can be pretty long. For example, this word [here](https://en.m.wiktionary.org/wiki/fast).

**"Just simple definitions?"**

- What would be a simple definition? I ask this because I'm not sure how the content for each grammatical category is structured in Wiktionary. This brings me to the next point.

**"All languages?"**

- For my use case, the ideal implementation would be to choose the languages (from the config/options of your plugin) that I'm interested in and only get the content from those. Or even better, select languages to include if available and languages to exclude.

Finally, thanks for this awesome plugin! It is essential for me, and I came to the issues asap after creating my GitHub account ^^

My motive for asking is if there is a way to automatically position the cursor or show the keyboard on mobile (not sure if this happens on PC) after using any of the features, especially for creating, linking, and searching. Maybe I can do it myself, but please @StrangeGirlMurph, let me know if that is possible.

**Comment by StrangeGirlMurph** [comment]
Hey! Thanks for joining us in this :) 
The main question really is: what tags (like `{url}`) are technically possible, which do you actually need and how to smoothly integrate the Wiktionary feature into a plugin that is used for the Wikipedia. I am pretty busy at the moment but I hopefully soon find some time to get a list of all the possible tags and think about the design choices for the integration. There are a lot of questions to be answered: Should the plugin be renamed? How to separate the wikipedia tags from the wiktionary? etc. 
Feel free to share your opinion on those! 

Regarding the other things you mentioned:

> My motive for asking is if there is a way to automatically position the cursor

I actually just created a new release that adds the `{cursor}` tag to dynamically set the position of the cursor after using "Link Article". I think that's exactly what you're looking for so go and update the plugin :)

> show the keyboard on mobile (not sure if this happens on PC) after using any of the features

Not sure what you mean by that. Please elaborate but first try again with the new version. I just used the plugin on mobile and after the insert I had no problems with the cursor or the keyboard not showing.

**Comment by codey-badger** [comment]
Hi,

I have updated the plugin and understand that smooth integration can be challenging, so no rush. Even just the hyperlinking of words could be useful for me.

Regarding the keyboard issue:

When I mentioned 'show the keyboard on mobile,' I meant that after using a command like `create article note`, my keyboard hides, and I have to manually tap into the search bar of the search pop-up to start typing. The same issue occurs with the `open article` command.

According to the documentation, in the section *Shared search modal*:
> If you had some text selected while calling one of the commands, your selection is automatically pasted into the search bar.

This functionality works perfectly with the `link article` command, but it does not work with the `open article` or `create article note` commands. This might be a bug; let me know if opening a new issue is appropriate.

**TL;DR**: 
**Expected behavior:**
- Keyboard remains visible after using create and search commands.
- Selection automatically pasted into the search bar after create and search commands (if possible).
- Let me know if I should create a new issue. 

**Device details:**
- Device: Samsung S9
- Android Version: 10
- Tested with two different keyboards: Samsung and Gboard
- Plugins installed: many, but only 3 enabled (dataview, calendar, and wiki search)

**Comment by StrangeGirlMurph** [comment]
Done with https://github.com/StrangeGirlMurph/obsidian-wikipedia-search/releases/tag/2.6.0 :)
